### PR TITLE
Add technote for manage and add 'manage' to keyword list in spec

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
Add technote about manage statement, add 'manage' keyword to spec

Write a technote about the `manage` statement that describes how it
works today, as well as its current status and open questions.

Add `manage` to the list of keywords in the "Lexical Structure"
section of the spec.

Reviewed by @arezaii and @lydia-duncan. Thanks!

Signed-off-by: David Longnecker <dlongnecke-cray@users.noreply.github.com>

### CODE REVIEW

**Comment by dlongnecke-cray** [comment]
Hi @lydia-duncan, would you be willing to do a quick pass over the technote for grammar/spelling/wording/wordsmithing purposes?

**Review by lydia-duncan — COMMENTED** [review]
I didn't notice any huge grammar or spelling errors.  I have a couple of suggestions that make more complete sentences and a couple of clarification suggestions

File: doc/rst/technotes/manage.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     proc leaveThis(in err: owned Error?) {
+       if err then halt(err:string);
+       writeln('x is: ', x);
+     }
+   }
+
+   var m = new myManager();
+   manage m as myResource {
+     // Prints '0'
+     myResource = 8;
```
Comment (cassella) [inline doc/rst/technotes/manage.rst:49]: `myResource.x = 8`?

Comment (dlongnecke-cray) [reply doc/rst/technotes/manage.rst:49]: Will do, thanks!

File: doc/rst/technotes/manage.rst [hunk 2]
```diff
@@ -16,16 +16,19 @@ keyword to open the statement instead of ``with``:
...
+The ``manage`` statement accepts a `manager` (the `myManager()` call
+in the above example). The statement calls a special method on the
+manager which lets it perform actions before executing the managed
+block.
 
 The manager may optionally return a `resource`. If a developer wants
 to make use of the resource, they may capture it after the manager
-expression (the `myResource` in the above example).
+expression (the `myResource` declaration in the above example).
+The resource may be any type, and is the value returned by the
```
Comment (lydia-duncan) [inline doc/rst/technotes/manage.rst:27]: Minor nit, but I think it should be "The resource may be of any type"?

**Review by lydia-duncan — APPROVED** [review]
Just one minor nit but I don't know if it's an absolutely necessary change.  Thanks!

**Pull request merged**